### PR TITLE
DOC: Added a section in the 'Iterating over arrays' doc page on basic iteration.

### PR DESCRIPTION
**Pull request by dpitch40**
It seems strange that basic iteration over arrays is covered only in the Quickstart Tutorial and not mentioned in the reference doc page titled "Iterating Over Arrays". I normally only use the reference section and spent about ten minutes trying to remind myself of how basic iteration works/wondering if it had been removed in an update. This pull request adds a section on basic iteration and moves everything that was there down one header level.

### CODE REVIEW

**Comment by rossbar** [comment]
Instead of explicitly duplicating "basic" iteration examples in the reference guide, how about linking to the relevant section of the tutorial instead? The advantage to such an approach is that it prevents fragmentation of the documentation, which can make things both hard to find for users and harder to maintain (see e.g. #14038)

File: doc/source/reference/arrays.nditer.cython.rst [hunk 1]
```diff
@@ -1,5 +1,5 @@
 Putting the Inner Loop in Cython
-================================
+--------------------------------
```
Comment (rossbar) [inline doc/source/reference/arrays.nditer.cython.rst:2]: No reason to change this - it shouldn't have any affect on the rendered html

**Review by rossbar — COMMENTED** [review]
I think going through and changing all of the heading styles makes this a more intrusive change than it needs to be. Rather than introducing a new heading, maybe the blurb you've added about "basic iteration" could be included as a `.. note` (or similar directive) at the top of the file?

File: doc/source/reference/arrays.nditer.rst [hunk 2]
```diff
@@ -10,6 +10,12 @@
 Iterating Over Arrays
 *********************
 
+.. note::
+
+   Arrays support the iterator protocol and can be iterated over like Python
+   lists. See the :ref:`quickstart.indexing-slicing-and-iterating` section in
+   the Quickstart guide for information and examples.
```
Comment (rossbar) [inline doc/source/reference/arrays.nditer.rst:17]:
```suggestion
   Arrays support the iterator protocol and can be iterated over like Python
   lists. See the :ref:`quickstart.indexing-slicing-and-iterating` section in
   the Quickstart guide for basic usage and examples. The remainder of
   this document presents the :class:`nditer` object and covers more 
   advanced usage.
```

**Review by rossbar — APPROVED** [review]
Thanks for sticking with this @dpitch40 , this is definitely an improvement IMO. I think you're right that the original document did not provide enough context and would be very confusing for newcomers who happened to stumble on this page looking for more basic usage/examples.

I've left a couple wording suggestions that you can feel free to use/ignore as you see fit (I think your original wording is fine) and took the liberty of pushing a minor `rST`-formatting nit to your branch.

**Review by melissawm — APPROVED** [review]
That makes sense, thank you @dpitch40 !

**Pull request merged**